kod/util/championsmaint.kod

```diff
@@ -0,0 +1,881 @@
+   piChampionsGame_stage_2_time = 1000 * 9 * TIME_FACTOR
+   piChampionsGame_stage_3_time = 1000 * 10 * TIME_FACTOR
+   piChampionsGame_stage_4_time = 1000 * 60 * 60 * 6 // wait 6 hours
+
+   // Provide multiple waves
+   piChampionsGame_wave_time = 0
+   piChampionsGame_wave_max_time = 1000 * 2 * TIME_FACTOR
+
+   // Rewards for winners and losers
+   // 2 waves at least for getting a reward
+   piChampionsGame_min_reward_time = 1000 * 2 * TIME_FACTOR * 2
+   piChampionsGame_reward_price = 2000
+   piChampionsGame_reward_winner = 5000 // 500
+   poChampionsGame_reward_loser = 1500 // 150
+   
+   // Lists
+   plChampions = $
+   
+messages:
+
+BoostChampion(who=$)
+{
+   local iBaseMaxHealth, iBoostHealth, iMaxHealth, iStamina;
+
+   if NOT pbChampionsGame_boost_champion_active
+   {
+      return;
+   }
+
+   iBaseMaxHealth = Send(who,@GetBaseMaxHealth);
+   iStamina = Send(who,@GetStamina);
+    
+   if iBaseMaxHealth <= (100+iStamina)
+   {
+      Send(who,@AddXP,#iAmount=10000000);
+      Send(who,@GivePlayerAllSkills);
+      Send(who,@GivePlayerAllSpells);
+      Send(who,@ConnectManaNode,#all=TRUE);
+   }
+
+   return;
+}
+
+CheckChampionsForRewardRequest()
+{
+   local i, iPoints, o, oChampion;
+
+   foreach o in plChampions
+   {
+      oChampion = Nth(o,1);
+      iPoints = Nth(o,2);
+      
+      if Send(self,@IsHoldingRewardItem,#who=oChampion)
+      {
+         if Send(self,@HasEnoughChampionPoints,#who=oChampion)
+         {
+            Send(self,@CreateItemReward,#who=oChampion);
+            Send(self,@UpdateChampionPoints,
+               #who=oChampion,
+               #what=-piChampionsGame_reward_price);
+            Send(oChampion,@MsgSendUser,
+               #message_rsc=champions_game_reward_successful_rsc,
+               #parm1=iPoints,
+               #parm2=piChampionsGame_reward_price);
+         }
+         else
+         {
+            Send(oChampion,@MsgSendUser,
+               #message_rsc=champions_game_reward_failed_rsc,
+               #parm1=iPoints,
+               #parm2=piChampionsGame_reward_price);
+         }
+      }
+   }
+
+   return;
+}
+
+Constructor(pvp=TRUE)
+{
+   ptChampionsGame_timer = CreateTimer(self,@Maintain,piChampionsGame_interval);
+   
+   Send(self,@CreateGame);
+   Send(self,@CreateRoom,#pvp=pvp);
+
+   return;
+}
+
+CreateGame()
+{
+   plChampions = [];
+
+   return;
+}
+
+CreateItemReward(who=$)
+{
+   local iTranslation, oClass, oItem, oRoom, oTarget, sInscription;
+
+   oItem = Send(self,@GetRewardItem,#who=who);
+   if oItem <> $
+   {
+      sInscription = Send(oItem,@GetInscription);
+      oClass = Send(self,@GetRewardItemClass,#sInscription=sInscription);
+      iTranslation = Send(self,@GetRewardItemTranslation,#sInscription=sInscription);
+      oTarget = Send(self,@GetRewardTarget,#who=who,#class=oClass);
+      if oTarget <> $
+      {
+         // Bug? item looks like enchanted (orange) but description will not appear on click
```

Comment:
The IA_BONDED attribute is not what players know of as 'bonded' or SWL. What you want here is IA_TRANSCENDANT which just needs `oItem` specified and not `oPlayer` and is the attribute used in the Bond spell/dm item bonding.

Reply:
great I didnt know how to solve that! I will change that